google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -2937,10 +2989,25 @@ def gapi_exists?
         def patch_gapi! *attributes
           return if attributes.empty?
           ensure_service!
+
           patch_args = attributes.to_h { |attr| [attr, @gapi.send(attr)] }
+
+          update_mode = nil
+          has_access_key = patch_args.key? :access
+          other_keys_exist = (patch_args.keys - [:access]).any?
+
+          if has_access_key && !other_keys_exist
```

Comment:
Is there any public-facing documentation for this that can be linked here in a comment?